lambdas/build-cri-oauth-request/src/main/java/uk/gov/di/ipv/core/buildcrioauthrequest/domain/CachedOAuthCriEncryptionKey.java

```diff
@@ -0,0 +1,12 @@
+package uk.gov.di.ipv.core.buildcrioauthrequest.domain;
+
+import com.nimbusds.jose.jwk.RSAKey;
+
+import java.util.Date;
+
+public record CachedOAuthCriEncryptionKey(RSAKey key, Date created) {
+    public Boolean isExpired(Integer cacheDurationMinutes) {
+        var expiryDate = new Date(created.getTime() + cacheDurationMinutes * 60 * 1000);
+        return expiryDate.before(new Date());
```

Comment:
You could simplify this by writing the `expiryTime` into the record, rather than the `createdTime` - it's a bit weird that this method takes an argument.